core/src/test/java/org/apache/iceberg/TestMergeAppend.java

```diff
@@ -92,6 +94,74 @@ public void testEmptyTableAppend() {
         statuses(Status.ADDED, Status.ADDED));
   }
 
+  @Test
+  public void testEmptyTableAppendFilesWithDifferentSpecs() {
+    assertThat(listManifestFiles().size()).as("Table should start empty").isEqualTo(0);
```

Comment:
```suggestion
    assertThat(listManifestFiles().isEmpty();
```
or `.hasSize(0)`. Both are preferred, because they will actually print the content of the collection if the assertion ever fails, whereas `assertThat(listManifestFiles().size()).isEqualTo(xyz)` will only say `3 != xyz` without knowing what the content is